### PR TITLE
Adding testing to the library

### PR DESCRIPTION
**Pull request by alsoto25**
# Description

We're looking into adding robust testing to the library to give all contributors more confidence that the changes being made are not breaking anything.

- Installed and added configuration for Playwright.
- Made a slight typing change to the configuration on the constructor to more accurately mark required and optional fields.
- Added base file structure for all tests we would ideally have.
- Added Github Action to run tests when a PR is created and when a PR is merged.

# Testing

The added tests were thoroughly tested using Playwright's UI:

image

### CODE REVIEW

File: tests/2-config.spec.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/*
```
Comment (KennethCalvo) [inline tests/2-config.spec.ts:1]: Maybe we can add these files until we have actual tests there.

**Review by KennethCalvo — APPROVED** [review]

**Review by Brian06 — APPROVED** [review]

**Review by mark13mark — APPROVED** [review]

**Review by wiremanb — APPROVED** [review]
LGTM

File: tests/1-basic.spec.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+declare global {
+  interface Window {
+    Validation: typeof Validation;
+  }
+}
+
+test.describe('Form Validation Basic Tests', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('http://127.0.0.1:3000/tests');
+  });
```
Comment (jdziat) [inline tests/1-basic.spec.ts:20]: Does playwright start its own web server? I thought it just automated a browser.

**Review by jdziat — COMMENTED** [review]
Just have a quick question about playwright

**Review by jdziat — APPROVED** [review]
LGTM

File: tests/2-config.spec.ts (hunk 1 above)
Comment (alsoto25) [reply tests/2-config.spec.ts:1]: The idea to adding these is so if other people come in and help, they don't have to take the decision on how to approach testing, which could lead to different people having different approaches in mind and then having to do extra work with the consolidation of tests and maybe conflicts.

This way each person could potentially just work on a single file without stepping on anyone elses' toes and the standard architecture of the tests is already there.

**Pull request merged**